cloudevents/tests/test_kafka_conversions.py

```diff
@@ -0,0 +1,517 @@
+        value=json.dumps(expected_data).encode("utf-8"),
+        key="test_key_123",
+    )
+
+
+@pytest.fixture
+def source_binary_bytes_message() -> ProtocolMessage:
+    return ProtocolMessage(
+        headers={
+            "ce_specversion": "1.0".encode("utf-8"),
+            "ce_id": "1234-1234-1234".encode("utf-8"),
+            "ce_source": "pytest".encode("utf-8"),
+            "ce_type": "com.pytest.test".encode("utf-8"),
+            "ce_time": datetime.datetime(2000, 1, 1, 6, 42, 33)
+            .isoformat()
+            .encode("utf-8"),
+            "content-type": "foo".encode("utf-8"),
+        },
+        value=_serialize(expected_data),
+        key="test_key_123",
+    )
+
+
+@pytest.fixture
+def source_structured_json_message() -> ProtocolMessage:
+    return ProtocolMessage(
+        headers={
+            "content-type": "foo".encode("utf-8"),
+        },
+        value=json.dumps(
+            {
+                "specversion": "1.0",
+                "id": "1234-1234-1234",
+                "source": "pytest",
+                "type": "com.pytest.test",
+                "time": datetime.datetime(2000, 1, 1, 6, 42, 33).isoformat(),
+                "data": expected_data,
+            }
+        ).encode("utf-8"),
+        key="test_key_123",
+    )
+
+
+@pytest.fixture
+def source_structured_json_bytes_message() -> ProtocolMessage:
+    return ProtocolMessage(
+        headers={
+            "content-type": "foo".encode("utf-8"),
+        },
+        value=json.dumps(
+            {
+                "specversion": "1.0",
+                "id": "1234-1234-1234",
+                "source": "pytest",
+                "type": "com.pytest.test",
+                "time": datetime.datetime(2000, 1, 1, 6, 42, 33).isoformat(),
+                "data_base64": base64.b64encode(_serialize(expected_data)).decode(
+                    "ascii"
+                ),
+            }
+        ).encode("utf-8"),
+        key="test_key_123",
+    )
+
+
+@pytest.fixture
+def source_structured_bytes_bytes_message() -> ProtocolMessage:
+    return ProtocolMessage(
+        headers={
+            "content-type": "foo".encode("utf-8"),
+        },
+        value=_serialize(
+            {
+                "specversion": "1.0",
+                "id": "1234-1234-1234",
+                "source": "pytest",
+                "type": "com.pytest.test",
+                "time": datetime.datetime(2000, 1, 1, 6, 42, 33).isoformat(),
+                "data_base64": base64.b64encode(_serialize(expected_data)).decode(
+                    "ascii"
+                ),
+            }
+        ),
+        key="test_key_123",
+    )
+
+
+@pytest.fixture
+def custom_marshaller() -> types.MarshallerType:
+    return _serialize
+
+
+@pytest.fixture
+def custom_unmarshaller() -> types.MarshallerType:
+    return _deserialize
+
+
+def _throw():
+    raise Exception("fail")
+
+
+@pytest.fixture
+def bad_marshaller() -> types.MarshallerType:
+    return _throw
+
+
```

Comment:
Can you please organize the tests created below under test classes? Probably one per method under a test.